src/Orchard.Environment.Commands/DefaultCommandManager.cs

```diff
@@ -67,7 +67,7 @@ private static IEnumerable<Match> MatchCommands(CommandParameters parameters, in
                     // We check here number of arguments a command can recieve against
                     // arguments provided for the command to identify the correct command
                     // and avoid matching multiple commands.
-                    if(commandDescriptor.MethodInfo.GetParameters().Length == argCount - names.Count())
+                    if (names[0] == parameters.Arguments.First() && commandDescriptor.MethodInfo.GetParameters().Length == argCount - names.Count())
```

Comment:
@sebastienros You are right, it is a bad fix I did just to fix the problem I found. It also doesn't take into account if a command name is composed by two terms.
My last commit fix it.